pkg/release/release.go

```diff
@@ -204,34 +204,26 @@ func (r *release) mirrorImages(envConfig *config.EnvConfig, applianceConfig *con
 		return err
 	}
 
-	p, err := os.Getwd()
+	absPath, err := filepath.Abs(templates.GetFilePathByTemplate(templateFile, envConfig.TempDir))
 	if err != nil {
 		return err
 	}
 
-	filePath := filepath.Join(p, templates.GetFilePathByTemplate(templateFile, envConfig.TempDir))
-	cmd := fmt.Sprintf(ocMirrorAndUpload, filePath, registry.RegistryPort)
-	logrus.Debugf("Fetching image from OCP release (%s)", cmd)
-
-	if err = syscall.Chdir(filepath.Join(p, envConfig.TempDir)); err != nil {
+	tempDir, err := ioutil.TempDir(envConfig.TempDir, "oc-mirror")
+	if err != nil {
 		return err
 	}
+	defer os.RemoveAll(tempDir)
+
+	cmd := fmt.Sprintf(ocMirrorAndUpload, absPath, registry.RegistryPort, tempDir)
+	logrus.Debugf("Fetching image from OCP release (%s)", cmd)
+
 	result, err := r.execute(r.executer, "", cmd)
 	logrus.Debugf("mirroring result: %s", result)
 	if err != nil {
 		return err
 	}
 
-	// remove oc mirror leftovers
```

Comment:
Why is this being removed?

Reply:
I mean, I get that this is related to what you did in above lines, but is it related to the PR $subject ?

Reply:
Yes, the change is required for monitoring the size of the mirror dir. I.e. it uses a relative path, so we can't use a syscall here to change the dir. Hence, had to change to logic a bit for displaying the progress correctly.